component/receiver.go

```diff
@@ -55,37 +55,6 @@ type LogsReceiver interface {
 	Receiver
 }
 
-// ReceiverFactoryBase defines the common functions for all receiver factories.
-type ReceiverFactoryBase interface {
-	Factory
-
-	// CreateDefaultConfig creates the default configuration for the Receiver.
-	// This method can be called multiple times depending on the pipeline
-	// configuration and should not cause side-effects that prevent the creation
-	// of multiple instances of the Receiver.
-	// The object returned by this method needs to pass the checks implemented by
-	// 'configcheck.ValidateConfig'. It is recommended to have such check in the
-	// tests of any implementation of the Factory interface.
-	CreateDefaultConfig() configmodels.Receiver
-}
-
-// ReceiverFactoryOld can create TraceReceiver and MetricsReceiver.
-type ReceiverFactoryOld interface {
```

Comment:
Does this break contrib components or we already removed all such usages from contrib?

Reply:
yes all contribs are converted :)

Reply:
Great!